tests/testthat/test-function_argument_linter.R

```diff
@@ -0,0 +1,9 @@
+test_that("function_argument_linter returns the correct linting", {
+  linter <- function_argument_linter()
+  msg <- rex::rex("Arguments without defaults should come before arguments with defaults.")
+
+  expect_lint("function(x, y = 1, z) {}", list(message = msg, column_number = 20L), linter)
```

Comment:
what happens with `...`? Add some tests & ensure `...` is handled correctly

Reply:
Added code and tests to ignore `...`, i.e. allow `function(a, b = 2, ...) {}`, but not `function(a, b = 2, ..., c)`.